openfisca_core/periods/date_unit.py

```diff
@@ -0,0 +1,259 @@
+from __future__ import annotations
+
```

Comment:
@benjello My idea to abstract date units from periods, so to facilitate their extension (weeks and so on), based on indexed enums. What do you think?